actionview/lib/action_view/template.rb

```diff
@@ -250,6 +250,14 @@ def encode!
       end
     end
 
+    # Build path relative to template location
+    def build_path(relative_path:)
+      return relative_path if !relative_path.start_with?(".")
```

Comment:
Why are we checking whether it's a relative path here again? The method signature indicates to pass a relative path.

It seems strange for a template (which has a definite location on disk) to calculate a path based on an outside parameter that points to the final disk location.

I'd have thought this would be something like:

```ruby
def template_path
  File.split(@virtual_path)
end
```

Reply:
I've renamed method to better reflect what it does

Reply:
> It seems strange for a template (which has a definite location on disk) to calculate a path based on an outside parameter that points to the final disk location.

The method name does sound clearer but ^ is still unaddressed.

Reply:
@kaspth I followed @tenderlove's suggestion here
> We should be asking the template for its file, or even better asking the template to figure out what the path should be. Like template.what_should_I_render_for_this_path("../../../whatever") (I think). I'm just spitballing on API here, but using vitual_path is definitely not it.